RHDEVDOCS-3942 Tracker for 2069771 - surface maven/nodejs pod templat… (#44220)
openshift/openshift-docs: modules/images-other-jenkins-config-kubernetes.adoc

Both the Maven and Node.js agent images are automatically configured as Kubernetes pod template images within the {product-title} Jenkins image configuration for the Kubernetes plug-in. That configuration includes labels for each of the images that can be applied to any of your Jenkins jobs under their Restrict where this project can be run setting. If the label is applied, jobs run under an {product-title} pod running the respective agent image.

[IMPORTANT]
====

> With this update, you can run Jenkins agents as sidecar containers. You can use this capability to run any container image in a Jenkins pipeline that has a correctly configured pod template and Jenkins file. Now, to compile code, you can run two new pod templates named `java-build` and `nodejs-builder` as sidecar containers with Jenkins. These two pod templates use the latest Java and NodeJS versions provided by the `java` and `nodejs` image streams in the `openshift` namespace.

Reviewer: ```suggestion
In {product-title} 4.10 and later, use of multi-container Kubernetes plug-in Pod Template as Jenkins agents, where one container is the 'jnlp' container and the other 'sidecar' container is a container whose image has the tools needed to build in a particular language, is now the recommended pattern. You can use this capability to run any container image from the Red Hat Container Catalog in a Jenkins pipeline that has a correctly configured pod template and Jenkins file. The {product-title} Jenkins image has two pod templates named `java-build` and `nodejs-builder` with sidecar containers that demonstrate this approach. These two pod templates use the latest Java and NodeJS versions provided by the `java` and `nodejs` image streams in the `openshift` namespace.
```

Author: Thanks. Updated.

> In {product-title} 4.10 and later, the recommended pattern for running Jenkins agents is to use the `jnlp` and `sidecar` containers as Kubernetes plug-in pod templates. The `sidecar` container image has the tools needed to build in a particular language. You can use this capability to run any container image from the Red Hat Container Catalog in a Jenkins pipeline that has a correctly configured pod template and Jenkins file. The {product-title} Jenkins image has two pod templates named `java-build` and `nodejs-builder` with sidecar containers that demonstrate this approach. These two pod templates use the latest Java and NodeJS versions provided by the `java` and `nodejs` image streams in the `openshift` namespace.

Author: Thank you! Update. WDYT now?

> In {product-title} 4.10 and later, the non-sidecar `maven` and `nodejs` pod templates for Jenkins are deprecated. These pod templates are planned for removal in a future release. Bug fixes and support are provided through the end of that future life cycle, after which no new feature enhancements are made. Instead, with this update, you can run Jenkins agents as sidecar containers.

Reviewer: ```suggestion
With this, in {product-title} 4.10 and later, the non-sidecar `maven` and `nodejs` pod templates for Jenkins are deprecated. These pod templates are planned for removal in a future release. Bug fixes and support are provided through the end of that future life cycle, after which no new feature enhancements are made. Instead, with this update, you can run Jenkins agents as sidecar containers.
```

Author: Thanks. Updated.

> With this, in {product-title} 4.10 and later, the non-sidecar `maven` and `nodejs` pod templates for Jenkins are deprecated. These pod templates are planned for removal in a future release. Bug fixes and support are provided through the end of that future life cycle, after which no new feature enhancements are made.

Reviewer 2: ```suggestion
With this update, in {product-title} 4.10 and later, the non-sidecar `maven` and `nodejs` pod templates for Jenkins are deprecated. These pod templates are planned for removal in a future release. Bug fixes and support are provided through the end of that future life cycle, after which no new feature enhancements are made.
```
- The antecedent of  `this` is unclear. "With this update"? "With this change"? "With this fix"?
